### PR TITLE
Added ping slash command (POC)

### PR DESCRIPTION
**Pull request by mcpenguin**
# Summary of Changes
Changed the `ping` command to a slash command as a proof of concept.

I used [this guide](https://www.sapphirejs.dev/docs/Guide/getting-started/creating-a-basic-app-command) for reference.

# Steps to Reproduce
1. Build and run the bot
2. Run `/ping`

## Expected output
image

# Notes
The biggest caveat for this PR is that this requires Sapphire v3, which is currently in beta. That being said, in the Sapphire server the people I talked to mentioned Sapphire v3 is stable and should be officially released soon.

### CODE REVIEW

File: src/commands/ping/ping.ts [hunk 1]
```diff
@@ -1,21 +1,28 @@
...
 })
 export class PingCommand extends Command {
-  async messageRun(message: Message): Promise<Message> {
-    const { client } = container;
-    const msg = await message.channel.send('Ping?');
-    const content = `Pong from JavaScript! Bot Latency ${Math.round(client.ws.ping)}ms. API Latency ${
-      msg.createdTimestamp - message.createdTimestamp
-    }ms.`;
-    return msg.edit(content);
+  public async chatInputRun(interaction: Command.ChatInputInteraction): Promise<APIMessage | Message<boolean>> {
```
Comment (alexz429) [inline src/commands/ping/ping.ts:18]: this removes the message command (with .ping). We should probs keep this at least until every command is slash.

**Review by alexz429 — CHANGES_REQUESTED** [review]

File: src/commands/ping/ping.ts (hunk 1 above)
Comment (mcpenguin) [reply src/commands/ping/ping.ts:18]: It seems like using slash commands makes regular commands not work.

**Review by alexz429 — CHANGES_REQUESTED** [review]

File: src/commands/ping/ping.ts (hunk 1 above)
Comment (mcpenguin) [reply src/commands/ping/ping.ts]: This is now done. I tried to abstract as much as possible - let me know what you think.

File: src/commands/ping/ping.ts [hunk 2]
```diff
@@ -1,21 +1,40 @@
...
+
+  const botLatency = client.ws.ping;
+  const apiLatency = initialMessageFromBotAsMessage.createdTimestamp - messageFromUserAsMessage.createdTimestamp;
+  const content = `Pong from JavaScript! Bot Latency ${botLatency}ms. API Latency ${apiLatency}ms.`;
+  return content;
+};
+
+export class PingCommand extends CodeyCommand {
+  messageWhenExecutingCommand = 'Ping?';
+  messageIfFailure = 'Failed to receive ping.';
```
Comment (alexz429) [inline src/commands/ping/ping.ts:24]: Good to make this an option, but I am hoping that the response should be a generic "codey backend error" that we can standardize in utils.

**Review by alexz429 — APPROVED** [review]
LGTM, some more utility optimizations can be implemented later / as we make use of this framework.

File: src/codeyCommand.ts [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+) => SapphireMessageResponse;
+
+export type CodeyCommandOptions = {
+  name: string;
+  aliases: string[];
+  description: string;
+  detailedDescription: string;
+};
+
+export class CodeyCommand extends SapphireCommand {
```
Comment (marko-polo-cheno) [inline src/codeyCommand.ts:22]: Hey Marcus! Everything looks awesome. 
Please just give a few more lines of comments/docs where things might be new/confusing so people can quickly change things for using slash-cmds

**Review by marko-polo-cheno — APPROVED** [review]

**Review by marko-polo-cheno — COMMENTED** [review]
oh also plz send in a screenshot of like a test or whatever to show that the ping slash cmds work

**Comment by mcpenguin** [comment]
image

**Pull request merged**